### PR TITLE
Add PublishRequest.Resources.

### PR DESCRIPTION
**Pull request by ericsnowcurrently**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/176/
Test PASSed.

File: csclient/params/params.go [hunk 1]
```diff
@@ -239,6 +239,7 @@ type PromulgateRequest struct {
 // See https://github.com/juju/charmstore/blob/v4/docs/API.md#put-idpublish
 type PublishRequest struct {
 	Published bool
+	Resources map[string]int `json:",omitempty"`
```
Comment (kat-co) [inline csclient/params/params.go]: Please comment with what the key-value is -- especially since it's a public property.

**Comment by kat-co** [comment]
LGTM with comment addressed.

**Comment by urosj** [comment]
If limitations are ok as discussed in IRC, then +1.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/177/
Test PASSed.

**Comment by ericsnowcurrently** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**